pkg/lca/seedgenerator.go

```diff
@@ -0,0 +1,294 @@
+			ObjectMeta: metaV1.ObjectMeta{
+				Name: name,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the seedgenerator is empty")
+
+		builder.errorMsg = "SeedGenerator name cannot be empty"
+	}
+
+	return &builder
+}
+
+// WithOptionsSeedGenerator creates seedgenerator with generic mutation options.
+func (builder *SeedGenerator) WithOptionsSeedGenerator(options ...SeedGeneratorAdditionalOptions) *SeedGenerator {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Setting seedgenerator additional options")
+
+	for _, option := range options {
+		if option != nil {
+			builder, err := option(builder)
+
+			if err != nil {
+				glog.V(100).Infof("Error occurred in mutation function")
+
+				builder.errorMsg = err.Error()
+
+				return builder
+			}
+		}
+	}
+
+	return builder
+}
+
+// CreateSeedGenerator makes a seedgenerator in the cluster and stores the created object in struct.
+func (builder *SeedGenerator) CreateSeedGenerator() (*SeedGenerator, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Creating the seedgenerator %s",
+		builder.Definition.Name)
+
+	var err error
+	if !builder.ExistsSeedGenerator() {
+		err = builder.apiClient.Create(context.TODO(), builder.Definition)
+		if err == nil {
+			builder.Object = builder.Definition
+		}
+	}
+
+	return builder, err
+}
+
+// PullSeedGenerator pulls existing seedgenerator from cluster.
+func PullSeedGenerator(apiClient *clients.Settings, name string) (*SeedGenerator, error) {
+	glog.V(100).Infof("Pulling existing seedgenerator name %s from cluster", name)
+
+	builder := SeedGenerator{
+		apiClient: apiClient,
+		Definition: &lcasgv1alpha1.SeedGenerator{
+			ObjectMeta: metaV1.ObjectMeta{
+				Name: name,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the seedgenerator is empty")
+
+		builder.errorMsg = "seedgenerator 'name' cannot be empty"
+	}
+
+	if !builder.ExistsSeedGenerator() {
+		return nil, fmt.Errorf("seedgenerator object %s doesn't exist", name)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// DeleteSeedGenerator removes the existing seedgenerator from a cluster.
+func (builder *SeedGenerator) DeleteSeedGenerator() (*SeedGenerator, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Deleting the seedgenerator %s",
+		builder.Definition.Name)
+
+	if !builder.ExistsSeedGenerator() {
+		return builder, fmt.Errorf("seedgenerator cannot be deleted because it does not exist")
+	}
+
+	err := builder.apiClient.Delete(context.TODO(), builder.Definition)
+
+	if err != nil {
+		return builder, fmt.Errorf("can not delete seedgenerator: %w", err)
+	}
+
+	builder.Object = nil
+
+	return builder, nil
+}
+
+// GetSeedGenerator returns seedgenerator object if found.
+func (builder *SeedGenerator) GetSeedGenerator() (*lcasgv1alpha1.SeedGenerator, error) {
```

Comment:
```suggestion
func (builder *SeedGenerator) Get() (*lcasgv1alpha1.SeedGenerator, error) {
```
This is a method so it doesn't have to be unique

Reply:
Updated. Thanks.